### PR TITLE
Add the Globalization Invariant mode tests

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @stephentoub @danmosemsft @weshaggard @krwq

File: src/System.Globalization/tests/Invariant/InvariantMode.cs [hunk 1]
```diff
@@ -0,0 +1,796 @@
...
+            foreach (var cul in s_cultureNames)
+            {
+                Assert.Equal(result, CultureInfo.GetCultureInfo(cul).CompareInfo.LastIndexOf(source, value, startIndex, count, options));
+                Assert.Equal(result, source.LastIndexOf(value, startIndex, count, GetStringComparison(options)));
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(IsPrefix_TestData))]
+
```
Comment (danmosemsft) [inline src/System.Globalization/tests/Invariant/InvariantMode.cs]: nit , some of your methods have newline between attribute and method

File: src/System.Globalization/tests/Invariant/InvariantMode.cs [hunk 2]
```diff
@@ -0,0 +1,796 @@
...
+        public void TestToUpper(string lower, string upper, bool result)
+        {
+            foreach (var cul in s_cultureNames)
+            {
+                Assert.Equal(result, CultureInfo.GetCultureInfo(cul).TextInfo.ToUpper(lower).Equals(upper, StringComparison.Ordinal));
+                Assert.Equal(result, lower.ToUpper().Equals(upper, StringComparison.Ordinal));
+            }
+        }
+
```
Comment (danmosemsft) [inline src/System.Globalization/tests/Invariant/InvariantMode.cs]: nit extra line

File: src/System.Globalization/tests/Invariant/Invariant.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <ProjectGuid>{9A8926D9-1D4C-4069-8965-A626F6CA8C29}</ProjectGuid>
+  </PropertyGroup>
+  <!-- Default configurations to help VS understand the configurations -->
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcoreapp-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcoreapp-Release|AnyCPU'" />
+  <PropertyGroup>
+    <!-- we are overriding the .json config file -->
```
Comment (stephentoub) [inline src/System.Globalization/tests/Invariant/Invariant.Tests.csproj]: Nit: it'd be good to include a "why" for this

File: src/System.Globalization/tests/Invariant/InvariantMode.cs [hunk 4]
```diff
@@ -0,0 +1,796 @@
...
+            return sc;
+        }
+
+        [Theory]
+        [MemberData(nameof(IndexOf_TestData))]
+
+        public void TestIndexOf(string source, string value, int startIndex, int count, CompareOptions options, int result)
+        {
+            foreach (var cul in s_cultureNames)
```
Comment (stephentoub) [inline src/System.Globalization/tests/Invariant/InvariantMode.cs]: Nit: var => string in this and other such cases

File: src/System.Globalization/tests/Invariant/InvariantMode.cs [hunk 5]
```diff
@@ -0,0 +1,796 @@
...
+        {
+            yield return new object[] { "en-US" };
+            yield return new object[] { "ja-JP" };
+            yield return new object[] { "fr-FR" };
+            yield return new object[] { "tr-TR" };
+            yield return new object[] { "" };
+        }
+
+        private static string[] s_cultureNames = new string[] { "en-US", "ja-JP", "fr-FR", "tr-TR", "" };
+
```
Comment (stephentoub) [inline src/System.Globalization/tests/Invariant/InvariantMode.cs]: Nit: readonly

File: src/System.Globalization/tests/Invariant/InvariantMode.cs [hunk 6]
```diff
@@ -0,0 +1,796 @@
...
+        public static IEnumerable<object[]> IndexOf_TestData()
+        {
```
Comment (stephentoub) [inline src/System.Globalization/tests/Invariant/InvariantMode.cs]: Don't we already have member data inputs for a bunch of these? Can we refactor them out of their current location and put them in a shared file that both projects can then just include?

Comment (tarekgh) [inline src/System.Globalization/tests/Invariant/InvariantMode.cs]: There is many differences compared to the data exist in other tests. this is the whole point for test invariant mode to ensure such differences.

**Pull request merged**